vignettes/setting_up_dm.Rmd

```diff
@@ -0,0 +1,107 @@
+vignette: >
+  %\VignetteIndexEntry{Preparing your data}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+<!-- README.md is generated from README.Rmd. Please edit that file -->
+
+This document describes how you can get your data into a `dm` object. Though this example is also built in already — just run `cdm_nycflights13()` — here, we'll go through the process step by step.
+
+The five tables we are working with contain information on flights from airports of the North-American city of New York to other destinations in the United States in 2013 and are available through the package [`nycflights13`](https://github.com/hadley/nycflights13)). `flights` represents the journeys taken by planes, `airlines` includes transport organizations' names (`name`) and their shortcuts (`carrier`), `airports` details the arrival and departure stations, `weater` the meterological context at each hour, while `planes` describes aircrafts' characteristics. Once we've loaded `nycflights13`, the aforementioned tables are all in our work environment, ready for being accessed.
+
+```{r}
+library(dm)
+library(nycflights13)
+```
+
+# Adding Tables
+
+First, we will tell `dm` what tables we want to work with and how they're connected. For this, we can use `dm` or `as_dm()`. 
+
+`as_dm()` accepts lists of objects — as of this version, you must explicitly name the objects.
+
+```{r}
+flights_dm_no_keys <- as_dm(
+  list(
+    "airlines" = airlines,
+    "airports" = airports,
+    "flights" = flights,
+    "planes" = planes,
+    "weather" = weather
+  )
+)
+```
+
+# Defining Keys
+
+Though you now have a new `dm` object, containing all your data, a key detail is missing still: How your five tables are connected the foreign keys) and which column(s) uniquely identify observation (the primary keys).
+
+## Primary Keys
+
+`dm` offers `cdm_enum_pk_candidates()` to identify primary keys and `cdm_add_pk()` to add them. 
+
+```{r}
+cdm_enum_pk_candidates(
+  dm = flights_dm_no_keys, 
+  table = planes
+  )
+```
+
+Now, add any primary keys you may have identified:
+
+```{r}
+flights_dm_only_pks <- flights_dm_no_keys %>%
+  cdm_add_pk(table = airlines, column = carrier) %>% 
+  cdm_add_pk(table = airports, column = faa) %>% 
+  cdm_add_pk(table = planes, column = tailnum)
+flights_dm_only_pks
+```
+
+To get an overview of the primary keys after setting them, call `cdm_get_all_pks()`.
+
+```{r}
+cdm_get_all_pks(flights_dm_only_pks)
+```
+
+## Foreign Keys
+
+```{r}
+cdm_enum_fk_candidates(
+  dm = flights_dm_only_pks,
+  table = flights,
+  ref_table = airlines
+)
+```
+
+
+You define how your tables relate using `cdm_add_fk()`. First, define the tables you wish to connect to another using the function's options `table` and `ref_table`. Then indicate in `column` which column of `table` refers to `ref_table`'s primary key, which you've defined above. Use `check = FALSE` to coerce a key. 
```

Comment:
I don't understand "coerce". `check = FALSE` omits the consistency checks, ideally we want a notion of weak relationships (#4).

Reply:
ah I must have misunderstood what it does. Is changed now.